airflow/models/dag.py

```diff
@@ -1051,14 +1050,14 @@ def _get_task_instances(
         if run_id:
             tis = tis.where(TaskInstance.run_id == run_id)
         if start_date:
-            tis = tis.where(DagRun.logical_date >= start_date)
+            tis = tis.where(DagRun.run_after >= start_date)
         if task_ids is not None:
             tis = tis.where(TaskInstance.ti_selector_condition(task_ids))
 
         # This allows allow_trigger_in_future config to take affect, rather than mandating exec_date <= UTC
         if end_date or not self.allow_future_exec_dates:
             end_date = end_date or timezone.utcnow()
-            tis = tis.where(DagRun.logical_date <= end_date)
+            tis = tis.where(DagRun.run_after <= end_date)
```

Comment:
Same.

Reply:
After the latest revert and testing, seems to work fine 🤔  resolve this one